src/main/kotlin/com/github/avrokotlin/avro4k/decoder/ListDecoder.kt

```diff
@@ -15,9 +16,11 @@ import org.apache.avro.generic.GenericArray
 import org.apache.avro.generic.GenericRecord
 
 @ExperimentalSerializationApi
-class ListDecoder(private val schema: Schema,
-                  private val array: List<Any?>,
-                  override val serializersModule: SerializersModule
+class ListDecoder(
+   private val schema: Schema,
+   private val array: List<Any?>,
+   override val serializersModule: SerializersModule,
+   private val namingStrategy: NamingStrategy,
```

Comment:
How about wrapping the namingStrategy in an (not yet existing) "AvroConfiguration" object? It would make it possible to easily introduce new configuration options and propagate them to all Decoder / Encoder.